hashcracker/client/lib/client.ex

```diff
@@ -0,0 +1,136 @@
+defmodule Client do
```

Comment:
just a suggestion: if the options list, the list of required options, and the various other option related lists were stored in module attributes, they could all be together here at the top of the module. This would tighten up the code below, and also make it easier to maintain the option list, as you could see how everything relates at a glance.